### PR TITLE
Feat: skip integrity check if offline

### PR DESCRIPTION
**Pull request by liyier90**
Closes https://github.com/aisingapore/PeekingDuck-Private/issues/93

- Skips weights integrity check and prints warning on `ConnectionErrror`
- Prints warning about possible connectivity issues when `input.visual` source is a URL
- Make `VideoThread` and `VideoNoThread` inherit from `VideoReader`
  - Abstract `read_frame()` and `queue_size` for the child classes to implement

### CODE REVIEW

File: tests/pipeline/nodes/input/test_input_threading.py [hunk 1]
```diff
@@ -54,23 +54,19 @@ def get_fps_number(avg_fps_msg: str) -> float:
...
+    except OSError as error:
+        if error.errno != errno.ENOENT:
+            # re-raise if exception isn't "no such file or directory"
+            raise
 
 
 # Unit Tests
+@pytest.mark.usefixtures("pipeline_instance")
 def test_input_threading():
     """Run input threading unit test.
```
Comment (ongtw) [inline tests/pipeline/nodes/input/test_input_threading.py:71]: Using the `pipeline_instance` fixture is a much better design!
Can update the docstring below as `1.` and `.4` are no longer valid/relevant.

**Review by ongtw — CHANGES_REQUESTED** [review]
Some minor changes before merge.

File: tests/pipeline/nodes/input/test_input_threading.py (hunk 1 above)
Comment (liyier90) [reply tests/pipeline/nodes/input/test_input_threading.py:71]: Updated docstring.

File: peekingduck/pipeline/nodes/input/utils/read.py [hunk 2]
```diff
@@ -29,16 +31,25 @@
...
 
-class VideoThread:
-    """
-    Videos will be threaded to improve FPS by reducing I/O blocking latency.
+def has_internet() -> bool:
+    """Checks for internet connectivity by making a HEAD request to one of
+    Google's public DNS servers.
     """
+    # Suppress bandit B309 as PeekingDuck is meant to run on Python >= 3.6
+    connection = http.client.HTTPSConnection("8.8.8.8", timeout=5)  # nosec
```
[the remark this answers is not in the document]
Comment (liyier90) [reply peekingduck/pipeline/nodes/input/utils/read.py]: declared `GOOGLE_DNS` constant

**Review by ongtw — APPROVED** [review]
LGTM

**Pull request merged**